shell/platform/windows/win32_window.cc

```diff
@@ -158,16 +158,59 @@ Win32Window::MessageHandler(HWND hwnd,
         tracking_mouse_leave_ = false;
         break;
       case WM_LBUTTONDOWN:
+        // Capture the pointer in case the user drags outside the client area.
+        // In this case, the "mouse leave" event is delayed until the user
+        // releases the button.
+        SetCapture(hwnd);
         xPos = GET_X_LPARAM(lparam);
         yPos = GET_Y_LPARAM(lparam);
         window->OnPointerDown(static_cast<double>(xPos),
-                              static_cast<double>(yPos));
+                              static_cast<double>(yPos), WM_LBUTTONDOWN);
         break;
       case WM_LBUTTONUP:
         xPos = GET_X_LPARAM(lparam);
         yPos = GET_Y_LPARAM(lparam);
         window->OnPointerUp(static_cast<double>(xPos),
-                            static_cast<double>(yPos));
+                            static_cast<double>(yPos), WM_LBUTTONUP);
+        ReleaseCapture();
+        break;
+      case WM_RBUTTONDOWN:
+        xPos = GET_X_LPARAM(lparam);
+        yPos = GET_Y_LPARAM(lparam);
+        window->OnPointerDown(static_cast<double>(xPos),
+                              static_cast<double>(yPos), WM_RBUTTONDOWN);
+        break;
+      case WM_RBUTTONUP:
+        xPos = GET_X_LPARAM(lparam);
+        yPos = GET_Y_LPARAM(lparam);
+        window->OnPointerUp(static_cast<double>(xPos),
+                            static_cast<double>(yPos), WM_RBUTTONUP);
+        break;
+      case WM_MBUTTONDOWN:
+        xPos = GET_X_LPARAM(lparam);
+        yPos = GET_Y_LPARAM(lparam);
+        window->OnPointerDown(static_cast<double>(xPos),
+                              static_cast<double>(yPos), WM_MBUTTONDOWN);
+        break;
+      case WM_MBUTTONUP:
+        xPos = GET_X_LPARAM(lparam);
+        yPos = GET_Y_LPARAM(lparam);
+        window->OnPointerUp(static_cast<double>(xPos),
+                            static_cast<double>(yPos), WM_MBUTTONUP);
+        break;
```

Comment:
Done. Added a note in the comment about the left click issue. I have no strong opinion on it so i'm happy to add other buttons if you think it's better.

Reply:
I don't have strong opinions either, but I do wonder if the cases you didn't add this handling for are going to assert in those edge cases.

Reply:
I just tested dragging out and releasing with other buttons and didn't get any assertions. I think we can leave it for now, since i'd still like to revisit this code and see if we should use the Raw Input api anyways